### ISSUE TITLE
problem accessing my datasource and/or API explorer after moving from LB 1.2 to 1.4

### ISSUE THREAD

**Issue by seanbrookes** [strongloop/loopback#142]
I use the following data source code to wire up my model definitions to my oracle db:

``` javascript
var nodeEnv = process.env.NODE_ENV || 'demo';
var cInst = require('rc')('loopback',{
  name: 'bacn-icar',
  env: nodeEnv
});
var config = cInst[nodeEnv]['oracle'];
var ds = loopback.createDataSource(config);
```

I recently updated LoopBack from 1.2 to 1.4 and my api explorer stopped working so I updated my app.js to try and get it working.  Now the explorer ui comes up but the rest calls are failing.  The response from the server for an api call looks like the following:

```
{
  "error": {
    "name": "TypeError",
    "status": 500,
    "message": "Object function ModelConstructor(data, dataSource) {\n            if(!(this instanceof ModelConstructor)) {\n                return new ModelConstructor(data, dataSource);\n            }\n            if(ModelClass.settings.unresolved) {\n                throw new Error('Model ' + ModelClass.modelName + ' is not defined.');\n            }\n            ModelBaseClass.apply(this, arguments);\n            if(dataSource) {\n                hiddenProperty(this, '__dataSource', dataSource);\n            }\n        } has no method 'find'",
    "stack": "TypeError: Object function ModelConstructor(data, dataSource) {\n            if(!(this instanceof ModelConstructor)) {\n                return new ModelConstructor(data, dataSource);\n            }\n            if(ModelClass.settings.unresolved) {\n                throw new Error('Model ' + ModelClass.modelName + ' is not defined.');\n            }\n            ModelBaseClass.apply(this, arguments);\n            if(dataSource) {\n                hiddenProperty(this, '__dataSource', dataSource);\n            }\n        } has no method 'find'\n    at Function.ACL.checkAccess (/Users/seanbrookes/_stacks/projects/bacn-icar/server/node_modules/loopback/lib/models/acl.js:366:7)\n    at Function.ACL.checkAccessForToken (/Users/seanbrookes/_stacks/projects/bacn-icar/server/node_modules/loopback/lib/models/acl.js:440:7)\n    at Function.Model.checkAccess (/Users/seanbrookes/_stacks/projects/bacn-icar/server/node_modules/loopback/lib/models/model.js:139:7)\n    at Function.app.enableAuth (/Users/seanbrookes/_stacks/projects/bacn-icar/server/node_modules/loopback/lib/application.js:168:13)\n    at execStack (/Users/seanbrookes/_stacks/projects/bacn-icar/server/node_modules/loopback/node_modules/strong-remoting/lib/remote-objects.js:285:11)\n    at RemoteObjects.execHooks (/Users/seanbrookes/_stacks/projects/bacn-icar/server/node_modules/loopback/node_modules/strong-remoting/lib/remote-objects.js:291:10)\n    at RemoteObjects.invokeMethodInContext (/Users/seanbrookes/_stacks/projects/bacn-icar/server/node_modules/loopback/node_modules/strong-remoting/lib/remote-objects.js:413:8)\n    at async.series.results (/Users/seanbrookes/_stacks/projects/bacn-icar/server/node_modules/loopback/node_modules/async/lib/async.js:548:21)\n    at _asyncMap (/Users/seanbrookes/_stacks/projects/bacn-icar/server/node_modules/loopback/node_modules/async/lib/async.js:224:13)\n    at async.eachSeries.iterate (/Users/seanbrookes/_stacks/projects/bacn-icar/server/node_modules/loopback/node_modules/async/lib/async.js:131:13)"
  }
}
```

stepping through the server code I and up in acl.js on line 366:

```
ACL.find({where: {model: model.modelName, property: propertyQuery,
    accessType: accessTypeQuery}}, function (err, acls) {
...
});
```

basically as follows:

```
throw new Error('Model ' + ModelClass.modelName + ' is not defined.');\n            }\n            ModelBaseClass.apply(this, arguments);\n            if(dataSource) {\n                hiddenProperty(this, '__dataSource', dataSource);\n            }\n        } has no method 'find'\n    at Function.ACL.checkAccess (/Users/seanbrookes/_stacks/projects/bacn-icar/server/node_modules/loopback/lib/models/acl.js:366:
```

My guess is some kind of ACL configuration is required...back to the docs.
